### PR TITLE
Fix #4987: Trackables might not be removed from a cache on refresh

### LINKED ISSUE #4987: Trackables might not be removed from a cache on refresh
2015.05.29-RC:

From IRC:
```
00:43:29OneEyed: if (hasBrand && removeTrackable) {
00:43:31OneEyed: iterator.remove();
00:43:33OneEyed: }
00:43:56OneEyed: Former trackables will never be removed if they don't have a brand originally
...
00:45:26OneEyed: kumy: am I correct that if there is a trackable of brand X, you refresh, and there are now zero trackables of brand X, it will never be removed?
00:45:42OneEyed: Geocache.java:1020
```

### PR DESCRIPTION
**Pull request by kumy**
Unit tests passed.
Do you imagine other test case?

### CODE REVIEW

**Comment by kumy** [comment]
@samueltardieu may you give me your comments please :)

File: main/src/cgeo/geocaching/Geocache.java [hunk 1]
```diff
@@ -1000,32 +1000,37 @@ public void setInventory(final List<Trackable> newInventory) {
...
+        while (iterator.hasNext()) {
+            final Trackable trackable = iterator.next();
+            boolean removeTrackable = true;
+            boolean hasBrand = false;
+            for (final Trackable newTrackable : newTrackables) {
+                if (trackable.getBrand() == newTrackable.getBrand()) {
+                    hasBrand = true;
+                    if (trackable.getUniqueID().equals(newTrackable.getUniqueID())) {
+                        removeTrackable = false;
+                        break;
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/Geocache.java]: Would not it be simpler to just start from a copy of `newTrackables`, and all all items in `inventory` whose brand is not in `processedBrands` (which could be made a set)?

Comment (kumy) [inline main/src/cgeo/geocaching/Geocache.java]: You're right.
Inverse the logic and things seems more simple ;)

**Comment by kumy** [comment]
PR updated

**Comment by kumy** [comment]
oups something is wrong :(

**Comment by kumy** [comment]
Fixed NPE and add unit test.
It's better now

**Comment by Lineflyer** [comment]
@samueltardieu Ready to merge this?

**Comment by samueltardieu** [comment]
I think it is. I can't do it though.

**Pull request merged**